wandb/sdk/internal/internal_api.py

```diff
@@ -3513,6 +3536,37 @@ def replace(a: str, b: str) -> None:
         link_artifact: Dict[str, Any] = response["linkArtifact"]
         return link_artifact
 
+    def fetch_org_entity_from_entity(self, entity: str) -> tuple[str, str]:
+        query = gql(
+            """
+            query FetchOrgEntityFromEntity(
+                $entityName: String!,
+            ) {
+                entity(name: $entityName) {
+                    organization {
+                        name
+                        orgEntity {
+                            name
+                        }
+                    }
+                }
+            }
+            """
+        )
+        response = self.gql(
+            query,
+            variable_values={
+                "entityName": entity,
+            },
+        )
+        print(response)
+        org = response.get("organization")
+        if not org or not org.get("orgEntity"):
+            raise ValueError(
+                f"Registry organization not found for entity '{entity}'"
+            )  # TODO: please use valid path
+        return org["orgEntity"]["name"] or "", org["name"] or ""
```

Comment:
suggestions:
 
- use try-except instead of `dict.get`, as long as we're confident in the response being a (nested) JSON object

- use `f"... {entity!r} ..."` instead of `f"... '{entity}' ..."`.  context: `{entity!r}` is shorthand for `{repr(entity)}` in an f-string, e.g.:
```python3
>>> x = None; print(f"got x: {x!r}")
got x: None
>>> x = "hello"; print(f"got x: {x!r}")
got x: 'hello'
```

```suggestion
        try:
            org = response["organization"]
            org_name = org["name"]
            org_entity_name = org["orgEntity"]["name"]
        except (LookupError, TypeError) as e:
            raise ValueError(
                f"Registry organization not found for entity {entity!r}"
            ) from e  # TODO: please use valid path
        else:
            return org_entity_name or "", org_name or ""
```

Reply:
thanks had no idea about the !r